internal/checker/checker.go

```diff
@@ -30482,6 +30483,36 @@ func (c *Checker) getRegularTypeOfExpression(expr *ast.Node) *Type {
 	return c.getRegularTypeOfLiteralType(c.getTypeOfExpression(expr))
 }
 
+func (c *Checker) containsArgumentsReference(node *ast.Node) bool {
+	links := c.nodeLinks.Get(node)
```

Comment:
Is there a more specific `links` that would make more sense here? I don't think we need to store this bit on every node's links, just functions, right?

Reply:
@jakebailey, thanks for the feedback. could the restriction apply only to declarations with a body, as that’s the starting point for traversal?

Reply:
It doesn't matter whether or not this is conditional because the field is still declared on the links type so will be paid for all node links. I was mainly wondering if this should go on `c.signatureLinks`.

Reply:
@jakebailey Yes, that makes sense. It would be cheaper. I initially followed the way TS handles it with `nodeLinks`